tests/remotes/gdrive.py

```diff
@@ -1,41 +1,85 @@
 # pylint:disable=abstract-method
+import locale
 import os
+import pathlib
+import tempfile
 import uuid
 
 import pytest
 from funcy import cached_property
 
-from dvc.path_info import CloudURLInfo
-from dvc.tree.gdrive import GDriveTree
+from dvc.path_info import CloudURLInfo, PathInfo
+from dvc.tree.gdrive import GDriveTree, GDriveURLInfo
 
 from .base import Base
 
 TEST_GDRIVE_REPO_BUCKET = "root"
 
 
 class GDrive(Base, CloudURLInfo):
+    def __init__(self, url, dvc):
+        CloudURLInfo.__init__(self, url)
+        Base.__init__(self, url)
+        self.tree = GDriveTree(dvc, self.config)
+
     @staticmethod
     def should_test():
         return bool(os.getenv(GDriveTree.GDRIVE_CREDENTIALS_DATA))
 
     @cached_property
     def config(self):
-        return {
-            "url": self.url,
-            "gdrive_service_account_email": "test",
-            "gdrive_service_account_p12_file_path": "test.p12",
-            "gdrive_use_service_account": True,
-        }
+        if GDrive.should_test():
+            return {
+                "url": self.url,
+            }
+        else:
+            return {
+                "url": self.url,
+                "gdrive_service_account_email": "test",
+                "gdrive_service_account_p12_file_path": "test.p12",
+                "gdrive_use_service_account": True,
+            }
 
     @staticmethod
     def _get_storagepath():
         return TEST_GDRIVE_REPO_BUCKET + "/" + str(uuid.uuid4())
 
+    @cached_property
+    def _get_gdrive_path_info(self):
+        return GDriveURLInfo(self.url)
+
```

Comment:
Not sure I understand the point of this, could you clarify, please? Why can't we just use `self` instead?

Reply:
because `self` is not an instance of `GDriveURLInfo`. So, it breaks some `assert` conditions.

Reply:
Ah, good point! I didn't realize that we didn't use it before here, it is by an accident. Does it break if we switch?

Reply:
Thanks for questioning again. In fact, the `assert` is not there anymore after I reverted some old changes. `self` works nicely.